typescript/packages/plugins/balancer/src/balancer.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Tool } from "@goat-sdk/core";
+import type { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { BalancerApi, ChainId, Slippage, SwapKind, Token, Swap, AddLiquidity, AddLiquidityKind, TokenAmount } from "@balancer/sdk";
+import { SwapParameters, LiquidityParameters } from "./parameters";
+
+export class BalancerService {
+    @Tool({
+        description: "Performs a token swap on Balancer using their Smart Order Router",
+    })
+    async swap(walletClient: EVMWalletClient, parameters: SwapParameters) {
+        const balancerApi = new BalancerApi(
```

Comment:
This can be done just once inside the constructor of the class and have `balancerApi` as a private field.

Reply:
done like thsi